tests/xharness/AppRunner.cs

```diff
@@ -365,8 +366,13 @@ public async Task<ProcessExecutionResult> UninstallAsync ()
 					// the right one (NUnitV3) add the nodes. ATM only TouchUnit uses V3.
 					var testRunName = $"{appName} {Variation}";
 					if (xmlType == XmlResultParser.Jargon.NUnitV3) {
+						var logs = new List<string> ();
+						// add our logs AND the logs of the previous task, which is the build task
+						logs.AddRange (Directory.GetFiles (Logs.Directory));
+						var previousLogsDir = Path.Combine (Directory.GetParent (Logs.Directory).FullName, BuildTaskID.ToString ());
```

Comment:
I think it would be better to store the BuildTask itself (instead of the ID), and then you could just do this here:

```csharp
var buildLogsDir = BuildTask.Logs.Directory;
```

which means you won't have to parse existing log structure (which may change in the future).